pysat/tests/test_instrument.py

```diff
@@ -615,6 +615,72 @@ def test_getting_all_data_by_numpy_array_of_int(self):
+                                         'mlt': 'mlt2'},
+                                        {'profiles': 'new_profiles'}])
+    def test_basic_variable_renaming(self, values):
+        # test single variable
+        self.testInst.load(2009, 1)
+        self.testInst.rename(values)
+        for key in values:
+            # check for new name
+            assert values[key] in self.testInst.data
+            assert values[key] in self.testInst.meta
+            # ensure old name not present
+            assert key not in self.testInst.data
+            assert key not in self.testInst.meta
+
+    def test_unknown_variable_error_renaming(self):
+        # check for error for unknown variable name
+        self.testInst.load(2009, 1)
+        with pytest.raises(ValueError):
+            self.testInst.rename({'help': 'I need somebody'})
+
+    @pytest.mark.parametrize("values", [{'profiles': {'density': 'ionization'}},
+                                        {'profiles': {'density': 'mass'},
+                                         'alt_profiles': {'density': 'volume'},
+                                         'alt_profiles': {'fraction': 'completion'}}])
+    def test_ho_pandas_variable_renaming(self, values):
+        # check for pysat_testing2D instrument
+        if self.testInst.platform == 'pysat':
+            if self.testInst.name == 'testing2D':
+                self.testInst.load(2009, 1)
+                self.testInst.rename(values)
+                for key in values:
+                    for ikey in values[key]:
+                        # check column name unchanged
+                        assert key in self.testInst.data
+                        assert key in self.testInst.meta
+                        # check for new name in HO data
+                        assert values[key][ikey] in self.testInst[0, key]
+                        check_var = self.testInst.meta[key]['children']
+                        assert values[key][ikey] in check_var
+                        # ensure old name not present
+                        assert ikey not in self.testInst[0, key]
+                        check_var = self.testInst.meta[key]['children']
+                        assert ikey not in check_var
+
+    def test_ho_pandas_unknown_variable_error_renaming(self):
+        # check for pysat_testing2D instrument
+        if self.testInst.platform == 'pysat':
+            if self.testInst.name == 'testing2D':
+                self.testInst.load(2009, 1)
+                # check for error for unknown HO variable name
+                with pytest.raises(ValueError):
+                    sub_dict = {'help': 'I need somebody'}
+                    self.testInst.rename({'profiles': sub_dict})
+                # check for bad column name when doing HO renaming
+                with pytest.raises(ValueError):
+                    sub_dict = {'help': 'I need somebody'}
+                    self.testInst.rename({'fake_profiles': sub_dict})
```

Comment:
This test can also be parametrized.

Reply:
Updated. Parameters for all! Tests pass locally.